dashboard/app/models/sections/omni_auth_section.rb

```diff
@@ -28,11 +28,13 @@
 
 class OmniAuthSection < Section
   def self.from_omniauth(code:, type:, owner_id:, students:, section_name: 'New Section')
-    oauth_section = with_deleted.where(code: code).first_or_create! do |section|
-      section.name = section_name || 'New Section'
-      section.user_id = owner_id
-      section.login_type = type
-    end
+    oauth_section = with_deleted.where(code: code).first_or_create
```

Comment:
`first_or_create!` will throw if it fails — is this behavior change intentional?

Reply:
Yes, because `first_or_create` only throws *after* the block is run, if it creates a new section, and the code that used to be in the block is necessary for the section to be valid. Now because we always want that code to run, we have to manually unpack the block and the save logic under the assumption that the `first_or_create` will not successfully save by itself.

Reply:
Note that the `save!` below will throw, which should make this equivalent in the end.